compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1830,6 +1830,10 @@ impl HasTargetSpec for Target {
     }
 }
 
+pub trait HasWasmCAbiOpt {
+    fn wasm_c_abi_opt(&self) -> bool;
+}
```

Comment:
This is the best I could come up with to keep the logic inside of `Target::adjust_abi()`, I believe moving this logic outside of it would introduce some very hacky code that could potentially become a trap for other contributors.

I'm pretty unfamiliar with the compiler though, so any advice would be appreciated.